plutus-contract/src/Wallet/Emulator/Wallet.hs

```diff
@@ -541,8 +541,8 @@ addInputs mp pk sk vl tx = do
     let
 
         addTxIns =
-            let ins = Set.fromList (Tx.pubKeyTxIn . fst <$> spend)
-            in over Tx.inputs (Set.union ins)
+            let ins = Tx.pubKeyTxIn . fst <$> spend
+            in over Tx.inputs ((++) ins)
```

Comment:
Hmm, I think we should use `Set` here to avoid duplicate inputs, convert to list and sort them.

@sjoerdvisscher what do you think?